### PR TITLE
Enable presubmit on Prow

### PR DESCRIPTION
**Pull request by sebastienvas**
- Fix Build Issues
- Create presubmit script
- Update e2e framework to save logs to _artifacts folder instead of uplading them to gcloud storage since Prow already does this.

### CODE REVIEW

File: prow/istio-presubmit.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+fi
+
+echo 'Running Linters'
+./bin/linters.sh
+
+echo 'Running Unit Tests'
+bazel test //...
+
+echo 'Running Integration Tests'
+./tests/e2e.sh --test_logs_path='_artifacts'
```
Comment (nlandolfi) [inline prow/istio-presubmit.sh]: I believe this should be `--test_log_path`, note no plural. based on the build log [here](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/464/istio-presubmit/3/)

**Comment by sebastienvas** [comment]
Prow results are available [here](https://storage.googleapis.com/istio-prow/pull/istio_istio/464/istio-presubmit/14/build-log.txt)

**Comment by sebastienvas** [comment]
Prow results are [here](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/464/istio-presubmit/17). Note that artifacts were also uploaded as well.

File: prow/istio-presubmit.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+fi
+
+echo 'Running Linters'
+./bin/linters.sh
+
+echo 'Running Unit Tests'
+bazel test //...
+
+echo 'Running Integration Tests'
+./tests/e2e.sh ${E2E_ARGS[@]}
```
Comment (nlandolfi) [inline prow/istio-presubmit.sh]: not as familiar with this syntax but basically this allows running the script locally? like you define the e2e args before the if for that purpose?

File: tests/e2e/framework/testInfo.go [hunk 3]
```diff
@@ -35,6 +36,7 @@ import (
 
 var (
 	logsBucketPath = flag.String("logs_bucket_path", "", "Cloud Storage Bucket path to use to store logs")
+	testLogsPath   = flag.String("test_log_path", "", "Local path to store logs in")
```
Comment (nlandolfi) [inline tests/e2e/framework/testInfo.go]: ahh i see the original typo came from that the var is testLogsPath whereas flag is test_log_path

File: prow/istio-presubmit.sh (hunk 2 above)
Comment (sebastienvas) [inline prow/istio-presubmit.sh]: Yes. Precisely.

File: tests/e2e/framework/testInfo.go (hunk 3 above)
Comment (sebastienvas) [inline tests/e2e/framework/testInfo.go]: I renamed it for consistency.

File: prow/istio-presubmit.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
```
Comment (ldemailly) [inline prow/istio-presubmit.sh]: a short description of what this does and how maybe would be a useful comment/header ?

File: prow/istio-presubmit.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+set -ex
+
+E2E_ARGS=()
+
+if [ "${CI}" == "bootstrap" ]; then
+    # ensure correct path
+    mkdir -p ${GOPATH}/src/istio.io
```
Comment (ldemailly) [inline prow/istio-presubmit.sh]: test if GOPATH is set first ?

File: prow/istio-presubmit.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    ln -s ${GOPATH}/src/github.com/istio/istio ${GOPATH}/src/istio.io
+    cd ${GOPATH}/src/istio.io/istio/
+    ARTIFACTS_DIR="${GOPATH}/src/istio.io/istio/_artifacts"
+    E2E_ARGS+=(--test_logs_path="${ARTIFACTS_DIR}")
```
Comment (ldemailly) [inline prow/istio-presubmit.sh]: I don't see an export or those 2 variables being used anywhere ?

File: tests/e2e/framework/testInfo.go [hunk 7]
```diff
@@ -61,24 +63,32 @@ type testStatus struct {
...
-	f := flag.Lookup("log_dir")
-	tmpDir := f.Value.String()
-	if tmpDir == "" {
-		var err error
-
-		tmpDir, err = ioutil.TempDir(os.TempDir(), tmpPrefix)
-		if err != nil {
+	tmpDir := ""
+	// testLogsPath will be used when called by Prow.
+	// Bootstrap already gather stdout and stdin so we don't need to keep the logs from glog.
```
Comment (ldemailly) [inline tests/e2e/framework/testInfo.go]: are we using -logtostderr for glog then ?

Comment (sebastienvas) [inline tests/e2e/framework/testInfo.go]: Yes. This is is in e2e.sh

File: prow/istio-presubmit.sh (hunk 6 above)
Comment (sebastienvas) [inline prow/istio-presubmit.sh]: E2E_ARGS is used down in the script.

File: prow/istio-presubmit.sh (hunk 5 above)
Comment (sebastienvas) [inline prow/istio-presubmit.sh]: GOPATH is set in the docker image. https://github.com/istio/test-infra/blob/master/docker/prowbazel/Dockerfile#L43

File: prow/istio-presubmit.sh (hunk 4 above)
Comment (sebastienvas) [inline prow/istio-presubmit.sh]: Done

File: prow/istio-presubmit.sh (hunk 5 above)
Comment (sebastienvas) [inline prow/istio-presubmit.sh]: Added set -u.

File: prow/istio-presubmit.sh (hunk 2 above)
Comment (ldemailly) [inline prow/istio-presubmit.sh]: oh I missed E2E_ARGS here (search fail!)

**Comment by nlandolfi** [comment]
bam!

**Pull request merged**